admin/includes/functions/general.php

```diff
@@ -738,23 +738,31 @@ function zen_set_ezpage_status(int $pages_id, int $status, string $status_field)
  */
 function zen_get_orders_status_pulldown_array()
 {
-    global $db;
+    $ordersStatus = zen_getOrdersStatuses();
+    return $ordersStatus['orders_statuses'];
+}
 
+function zen_getOrdersStatuses(bool $keyed = false): array
+{
+    global $db;
+    $orders_statuses = [];
     $orders_status_array = [];
-    $orders_status = $db->Execute("SELECT orders_status_id, orders_status_name
-                                   FROM " . TABLE_ORDERS_STATUS . "
-                                   WHERE language_id = " . (int)$_SESSION['languages_id'] . "
-                                   ORDER BY orders_status_id");
-
-    while (!$orders_status->EOF) {
-        $orders_status_array[] = [
-            'id' => $orders_status->fields['orders_status_id'],
-            'text' => $orders_status->fields['orders_status_name']
-        ];
-        $orders_status->MoveNext();
+    $orders_status_query = $db->Execute('SELECT orders_status_id, orders_status_name FROM ' . TABLE_ORDERS_STATUS . '
+                                 WHERE language_id = "' . (int)$_SESSION['languages_id'] . '" ORDER BY sort_order, orders_status_id');
+    while (!$orders_status_query->EOF) {
+        if (!$keyed) {
+            $orders_statuses[] = [
+                'id' => $orders_status_query->fields['orders_status_id'],
+                'text' => $orders_status_query->fields['orders_status_name'] . ' [' . $orders_status_query->fields['orders_status_id'] . ']',
+            ];
+            $orders_status_array[$orders_status_query->fields['orders_status_id']] = $orders_status_query->fields['orders_status_name'] . ' [' . $orders_status_query->fields['orders_status_id'] . ']';
+        } else {
+            $orders_statuses[$orders_status_query->fields['orders_status_id']] = $orders_status_query->fields['orders_status_name'];
+            $orders_status_array[$orders_status_query->fields['orders_status_id']] = $orders_status_query->fields['orders_status_name'];
+        }
+        $orders_status_query->MoveNext();
```

Comment:
Use `foreach ($orders_status_query as $next_status) {` instead of the `while/MoveNext()` loop, replacing each instance of `$orders_status_query->fields` with `$next_status`.